### ISSUE TITLE
Monitoring Server verify_and_map Exception?

### ISSUE THREAD

**Issue by kozlovic** [nats-io/nats-server#2980]
Originally created as a discussion by @rsoberano-ld

From the nats-server [tls config documentation](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/tls) about the verify_and_map option:

```
If true, require and verify client certificates and map certificate values for authentication purposes. Does not apply to monitoring either.
```

This seems to no longer work as expected after upgrading to version 2.6.6 (still present in 2.7.4); browser connections to the monitoring server on port 8222 seem to fail because the browser isn't presenting a client certificate. Example error:

```
This site can’t provide a secure connectionlocalhost didn’t accept your login certificate, or one may not have been provided.
Try contacting the system admin.
ERR_BAD_SSL_CLIENT_AUTH_CERT
```

TLS config for reference: 

```
host: 0.0.0.0
port: 4222
syslog: False
https_port: 8222

tls {
  cert_file: /etc/ssl/gnatsd/gnatsd-server.crt
  key_file: /etc/ssl/gnatsd/gnatsd-server.key
  ca_file: /etc/ssl/gnatsd/ca.crt
  verify_and_map: true
}
```

I haven't seen any other reports about the client cert exception breaking after v2.6.6 though, is it something specific to our config that's causing this behavior? 

Alternatively, looking at the changes to 2.6.6 in server/server-go:

```		
config.GetConfigForClient = s.getTLSConfig
config.ClientAuth = tls.NoClientCert
```

Since GetConfigForClient is refreshing the TLS config after a ClientHello, is it possibly overwriting the `config.ClientAuth = tls.NoClientCert` in the line below?

**Comment by kozlovic** [comment]
@rsoberano-ld I believe that this has been addressed now in PR #2981 that has been merged to main and should be available with nightly builds.

**Comment by rsoberano-ld** [comment]
Awesome, really appreciate the quick fix here.
